File: Ywy/JavaScript/level1/WeirdStringMake.js
export default class WeirdStringMake {
    solution = s => {
        let index = 0;
        return [...s].map(at => {
            if (at == " ") {
                index = 0;
                return at;
            }
            return index++ % 2 === 0 ? at.toUpperCase() : at.toLowerCase();
        }).join("");
    };

    solution2 = s => s.split(" ").map(str => [...str].map((at, index) => index % 2 > 0 ? at.toLowerCase() : at.toUpperCase())).join(" ").replace(/,/g, "");

} 

File: Ywy/JAVA/src/level1/WeirdStringMake.java
package level1;

public class WeirdStringMake {
	
	public WeirdStringMake() {}
	
	public String solution(final String s) {
		final StringBuffer BUFFER = new StringBuffer();
		String[] strArr = s.split(" ", -1);
		int index = 0;
		int range = strArr.length;
		while(true) {
			change(BUFFER, strArr[index++]);
			if (index == range) {
				return BUFFER.toString();
			}
			BUFFER.append(" ");
		}
	}
	
	private void change(final StringBuffer BUFFER, final String s) {
		int CHANGE_VALUE = 'A' / 2;
		char[] atArr = s.toCharArray();
		int index = 0;
		int range = atArr.length;
		char temp = '\u0000';
		while(index < range) {
			temp = atArr[index];
			temp += index % 2 == 0 ? -CHANGE_VALUE : +CHANGE_VALUE;
			if (temp > 64 && temp < 123) {
				atArr[index] = temp;
			}
			index++;
		}
		BUFFER.append(atArr);
	}
}
